tools/clang/test/HLSLFileCheck/validation/bitfield_swizzle.hlsl

```diff
@@ -0,0 +1,31 @@
+// RUN: %dxc -T ps_6_6 -HV 2021 %s | FileCheck %s
+
+// This test is a regression test that tests the implementation of L-Value
+// conversions work on bitfields, for both constant and structured buffers
+
+#define USE_BIT_FIELD
+
+struct MyStruct
+{
+#ifdef USE_BIT_FIELD
+    uint v0: 5;
+    uint v1: 15;
+    uint v2: 12;
+#else
+    uint v2;
+#endif
+};
+
+ConstantBuffer<MyStruct> myConstBuff;
+StructuredBuffer<MyStruct> myStructBuff;
+
+// CHECK: extractvalue %dx.types.ResRet.i32
+// CHECK: extractvalue %dx.types.CBufRet.i32
+
+
+[RootSignature("RootFlags(0), DescriptorTable(CBV(b0), SRV(t0))")]
+float4 main(uint addr: TEXCOORD): SV_Target
+{
+    // return myConstBuff.v2.x; // works fine with bitfields
+    return myStructBuff[0].v2.xxxx + myConstBuff.v2.xxxx; // crash with bitfields. Using a structured buffer instead of a constant buffer also crashes   
```

Comment:
I don't think you meant to leave this comment here.

For more exhaustive testing, can we get tests for some other swizzles even if they are invalid? .xyyz for example?